pkg/policy/resolve_deny_test.go

```diff
@@ -0,0 +1,513 @@
+// Copyright 2018-2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !privileged_tests
+
+package policy
+
+import (
+	"github.com/cilium/cilium/pkg/checker"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/identity/cache"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/policy/api"
+	"github.com/cilium/cilium/pkg/policy/trafficdirection"
+
+	. "gopkg.in/check.v1"
+)
+
+func GenerateL3IngressDenyRules(numRules int) api.Rules {
+	parseFooLabel := labels.ParseSelectLabel("k8s:foo")
+	fooSelector := api.NewESFromLabels(parseFooLabel)
+	barSelector := api.NewESFromLabels(labels.ParseSelectLabel("bar"))
+
+	// Change ingRule and rule in the for-loop below to change what type of rules
+	// are added into the policy repository.
+	ingDenyRule := api.IngressDenyRule{
+		IngressCommonRule: api.IngressCommonRule{
+			FromEndpoints: []api.EndpointSelector{barSelector},
+		},
+	}
+
+	var rules api.Rules
+	for i := 1; i <= numRules; i++ {
+
+		rule := api.Rule{
+			EndpointSelector: fooSelector,
+			IngressDeny:      []api.IngressDenyRule{ingDenyRule},
+		}
+		rule.Sanitize()
+		rules = append(rules, &rule)
+	}
```

Comment:
```suggestion
	rules := make([]api.Rules, 0, numRules)
	for i := 1; i <= numRules; i++ {
		rule := api.Rule{
			EndpointSelector: fooSelector,
			IngressDeny:      []api.IngressDenyRule{ingDenyRule},
		}
		rule.Sanitize()
		rules = append(rules, &rule)
	}
```

Reply:
I see that this loop is repeated, so might be worth extracting out.

Reply:
One is for ingress the other for egress.